src/Parallel/Callback.hpp

```diff
@@ -6,15 +6,45 @@
 
 #pragma once
 
+#include <optional>
 #include <pup.h>
 #include <tuple>
 #include <utility>
 
 #include "Parallel/Invoke.hpp"
+#include "Utilities/PrettyType.hpp"
 #include "Utilities/Serialization/CharmPupable.hpp"
 #include "Utilities/Serialization/RegisterDerivedClassesWithCharm.hpp"
+#include "Utilities/TypeTraits/HasEquivalence.hpp"
 
 namespace Parallel {
+namespace detail {
+// Not all tuple arguments are guaranteed to have operator==, so we check the
+// ones we can.
```

Comment:
I'm a bit worried about silently ignoring entries.  What types typically occur that cause problems?

Reply:
The problematic types were proxies and I didn't want to deal with how to evaluate equivalence of those

Reply:
OK.  https://github.com/charmplusplus/charm/pull/3848 is probably the real fix, for what it's worth, but we can't wait for that.